# PROJECT
## Applications of Big Data

### 0. Importing libraries

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.gridspec as gridspec

### 2. Data Understanding

#### 2.1. Tổng quan kích thước và kiểu dữ liệu trong từng cột

In [ ]:
df = pd.read_csv('dataset/data_clean.csv')
df.head()

Job Id Qualifications    location           Country  Work Type  \
0  1089843540111562         M.Tech     Douglas       Isle of Man     Intern   
1   398454096642776            BCA    Ashgabat      Turkmenistan     Intern   
2   481640072963533            PhD       Macao  Macao SAR, China  Temporary   
3   688192671473044            PhD  Porto-Novo             Benin  Full-Time   
4   117057806156508            MBA    Santiago             Chile     Intern   

   Company Size Job Posting Date Preference                     Job Title  \
0         26801       2022-04-24     Female  Digital Marketing Specialist   
1        100340       2022-12-19     Female                 Web Developer   
2         84525       2022-09-14       Male            Operations Manager   
3        129896       2023-02-25     Female              Network Engineer   
4         53944       2022-10-11     Female                 Event Manager   

                        Role  ...  \
0       Social Media Manager  ...   
1     Frontend Web Developer  ...   
2    Quality Control Manager  ...   
3  Wireless Network Engineer  ...   
4         Conference Manager  ...   

                                     Job Description  \
0  Social Media Managers oversee an organizations...   
1  Frontend Web Developers design and implement u...   
2  Quality Control Managers establish and enforce...   
3  Wireless Network Engineers design, implement, ...   
4  A Conference Manager coordinates and manages c...   

                                            Benefits  \
0  {'Flexible Spending Accounts (FSAs), Relocatio...   
1  {'Health Insurance, Retirement Plans, Paid Tim...   
2  {'Legal Assistance, Bonuses and Incentive Prog...   
3  {'Transportation Benefits, Professional Develo...   
4  {'Flexible Spending Accounts (FSAs), Relocatio...   

                                              skills  \
0  Social media platforms (e.g., Facebook, Twitte...   
1  HTML, CSS, JavaScript Frontend frameworks (e.g...   
2  Quality control processes and methodologies St...   
3  Wireless network design and architecture Wi-Fi...   
4  Event planning Conference logistics Budget man...   

                                    Responsibilities  \
0  Manage and grow social media accounts, create ...   
1  Design and code user interfaces for websites, ...   
2  Establish and enforce quality control standard...   
3  Design, configure, and optimize wireless netwo...   
4  Specialize in conference and convention planni...   

                            Company Salary_avg  Experience_avg  Day  Month  \
0                 Icahn Enterprises       79.0            10.0   24      4   
1      PNC Financial Services Group       86.0             7.0   19     12   
2  United Services Automobile Assn.       82.5             6.0   14      9   
3                              Hess       78.0             7.5   25      2   
4                      Cairn Energy       75.5             6.5   11     10   

   Year  
0  2022  
1  2022  
2  2022  
3  2023  
4  2022  

[5 rows x 21 columns]

In [ ]:
df.shape

(1615940, 21)

Dữ liệu có 1,615,940 dòng và 23 cột. Và dữ liệu được thu thập từ ngày `15/09/2021` đến ngày `15/09/2023`.

Ý nghĩa của một số cột như sau (Tương tự như ở phần `Data Processing`):

- Qualifications: Trình độ học vấn
- Experience_avg: Kinh nghiệm
- Salary_avg_: Mức lương
- Job Postinging Date: Ngày đăng tuyển dụng
- Company Size: Quy mô công ty
- Role: Vị trí công việc

Ngoài ra còn có các cột khác như: Job Title, Country, location, Contact Person, Contact, Company Profile, Industry,... được mô tả ở `Data_description.txt`. 

In [ ]:
print(df.info());

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1615940 entries, 0 to 1615939
Data columns (total 21 columns):
 #   Column            Non-Null Count    Dtype  
---  ------            --------------    -----  
 0   Job Id            1615940 non-null  int64  
 1   Qualifications    1615940 non-null  object 
 2   location          1615940 non-null  object 
 3   Country           1615940 non-null  object 
 4   Work Type         1615940 non-null  object 
 5   Company Size      1615940 non-null  int64  
 6   Job Posting Date  1615940 non-null  object 
 7   Preference        1615940 non-null  object 
 8   Job Title         1615940 non-null  object 
 9   Role              1615940 non-null  object 
 10  Job Portal        1615940 non-null  object 
 11  Job Description   1615940 non-null  object 
 12  Benefits          1615940 non-null  object 
 13  skills            1615940 non-null  object 
 14  Responsibilities  1615940 non-null  object 
 15  Company           1615940 non-null  object 
 16  

Dữ liệu của các cột:

- Job ID, Experience_avg, Salary_avg, Company Size: là kiểu numerical.
- Cột Job Posting Date: là kiểu datetime và các cột Day, Month, Year là kiểu numerical được trích từ cột Job Posting Date.
- Còn lại là kiểu categorical.


In [ ]:
df['Year'].value_counts()

2022    806158
2023    570838
2021    238944
Name: Year, dtype: int64

Qua đó ta thấy dữ liệu được thu thập trong 3 năm từ 2021 đến 2023.

In [ ]:
df1 = df['Country'].unique()
df1 = pd.DataFrame(df1)
df1.sort_values(by=0, inplace=True)
df1

0
58             Afghanistan
210                Albania
205                Algeria
131         American Samoa
158                Andorra
..                     ...
38   Virgin Islands (U.S.)
34      West Bank and Gaza
15                   Yemen
100                 Zambia
157               Zimbabwe

[216 rows x 1 columns]

Ở đây ta thấy có 216 quốc gia và khi nhóm tìm hiểu thì tính tới 2023 theo LHQ thì trên thới giới chỉ có 204 nước và lãnh thổ(`sputniknews.vn`). Tuy nhiên qua kiểm tra thì không có quốc gia nào trùng nhau và có một số quốc gia có quốc kỳ tuy nhiên vẫn thuộc một quốc gia khác. Ví dụ `Virgin Islands (U.S.)`, `Bahamas`,... đều thuộc Hoa Kỳ nên ở đây nhóm sẽ giữ nguyên. Vì ở đây có thể là do đã tính luôn cả các vùng, các quốc gia tự trị hoặc lãnh thổ đặc biệt là các `quốc gia` riêng biệt.

In [ ]:
df['Country'].value_counts()

Malta            7723
Venezuela, RB    7694
Somalia          7689
Sierra Leone     7656
Vietnam          7649
                 ... 
Ghana            7312
Ukraine          7307
Iceland          7292
Uganda           7290
Lithuania        7273
Name: Country, Length: 216, dtype: int64

Qua dữ liệu thì nhu cầu tuyển dụng của các quốc gia trên thế giới trong 3 năm 2021-2023 đều vượt qua 7200 công việc, trong đó `Việt Nam` nằm  trong top 5 quốc gia có như cầu tuyển dụng cao nhất thế giới. do đó ta có thể thấy nhu cầu tuyển dụng của `Việt Nam` là rất cao.

In [ ]:
df[['location', 'Country']].nunique()

location    214
Country     216
dtype: int64

Sau khi tìm hiểu thì theo dữ leieuj `Seoul` thì nó nằm ở `Korea, Rep.` và `North Korea` và `Apia` là thủ đô cũng như thành phố lớn nhất tại `Samoa` nhưng theo dữ liệu thì nó thêm cả `American Samoa`. Vì vậy nhóm sẽ giữ nguyên.